### PR TITLE
Support CBOR

### PR DESCRIPTION
**Pull request by paulgb**

### CODE REVIEW

File: js-pkg/driftdb/src/index.ts [hunk 1]
```diff
@@ -29,7 +30,7 @@ export class SubscriptionManager<T> {
     subscriptions: Map<string, EventListener<T>> = new Map()
 
     subscribe(subject: Key, listener: (event: T) => void) {
-        const key = JSON.stringify(subject)
+        const key = subject
```
Comment (rolyatmax) [inline js-pkg/driftdb/src/index.ts:33]: Maybe just rename the `subject` arg to `key` like you did in `dispatch()`?

File: js-pkg/driftdb/src/index.ts [hunk 2]
```diff
@@ -115,18 +122,25 @@ export class DbConnection {
...
             }, 1000)
         }
 
         this.connection.onmessage = (event) => {
-            const message: MessageFromDb = JSON.parse(event.data as string)
+            let message: MessageFromDb
+            if (event.data instanceof ArrayBuffer) {
+                message = CBOR.decode(event.data as any)
+            } else {
+                message = JSON.parse(event.data as string)
```
Comment (rolyatmax) [inline js-pkg/driftdb/src/index.ts:134]: Do you need the `as any` and `as string` here?

File: js-pkg/tests/src/driftdb.test.ts [hunk 3]
```diff
@@ -39,18 +38,17 @@ class CallbackExpecter<T> {
...
     }
 }
 
-async function connectToNewRoom(): Promise<{ db: DbConnection, room: RoomResult }> {
+async function connectToNewRoom({binary}: {binary?: boolean} = {}): Promise<{ db: DbConnection, room: RoomResult }> {
     let api = new Api(API_SERVER);
     let room = await api.newRoom();
     let db = new DbConnection();
-    await db.connect(room.socket_url);
+    await db.connect(room.socket_url, binary);
```
Comment (rolyatmax) [inline js-pkg/tests/src/driftdb.test.ts:51]: As far as I understand it, `db.send()` takes some JavaScript values whether or not `binary=true`. The only difference is that if your JavaScript values include TypedArrays or an ArrayBuffer, you need to set `binary=true` for it to work. So what's the point of having `binary=false`? Does it make sense to support JSON at all?

And if there is some use case where developers need to know about the transport format of the messages (and it's not just some implementation detail), then might they care about what format is actually being used, i.e. this isn't just binary, it's specifically CBOR.

I guess the TLDR is: Maybe we should just always use CBOR for serialization and not give the developer any options for the transport format. _Or_ if we _do_ think developers should be able to control the transport format, then maybe the options should be JSON vs CBOR (possibly with CBOR being the default?).

File: js-pkg/driftdb/src/index.ts (hunk 1 above)
Comment (paulgb) [reply js-pkg/driftdb/src/index.ts]: good catch

File: js-pkg/driftdb/src/index.ts (hunk 2 above)
Comment (paulgb) [reply js-pkg/driftdb/src/index.ts]: done

File: driftdb-server/src/server.rs [hunk 4]
```diff
@@ -63,18 +66,29 @@ impl<Inbound: DeserializeOwned, Outbound: Serialize> TypedWebSocket<Inbound, Out
...
+                .send(axum::extract::ws::Message::Binary(writer))
+                .await?;
+            return Ok(());
+        } else {
+            let msg = serde_json::to_string(&msg)?;
+
+            self.socket
             .send(axum::extract::ws::Message::Text(msg))
             .await?;
+        }
```
[the remark this answers is not in the document]
Comment (paulgb) [reply driftdb-server/src/server.rs:85]: done

**Review by rolyatmax — APPROVED** [review]

**Pull request merged**